noir-projects/noir-protocol-circuits/crates/types/src/utils/arrays.nr

```diff
@@ -65,12 +65,14 @@ where
     BoundedVec::from_parts_unchecked(array, len)
 }
 
+// Helper function to find the index of the first element in an array that satisfies a given predicate.
```

Comment:
Also mention that the return value will equal `N` if no element is found. And maybe yell at the kernel team for not using `Option`.

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/11094/commits/b949fcee7e996e1ce669e1f7597d9374dee6e608